lib/core/include/irods/authenticate.h

```diff
@@ -0,0 +1,70 @@
+#ifndef IRODS_AUTHENTICATE_H
+#define IRODS_AUTHENTICATE_H
+
+/// \file
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct RcComm;
+
+// These macros are used extensively throughout the authentication code. Over time, we should move away from using
+// these, but we cannot escape from it for now. Just know that these values are currently very important.
+#define MAX_PASSWORD_LEN 50 // NOLINT(cppcoreguidelines-macro-usage)
+#define CHALLENGE_LEN    64 // NOLINT(cppcoreguidelines-macro-usage)
+#define RESPONSE_LEN     16 // NOLINT(cppcoreguidelines-macro-usage)
```

Comment:
Hm... I should probably make these `NOLINTBEGIN`/`NOLINTEND`. Looks weird next to a macro.

Reply:
Done